src/main/java/com/npsproject/it4045npsproject/service/IParkService.java

```diff
@@ -0,0 +1,50 @@
+package com.npsproject.it4045npsproject.service;
+
+import com.npsproject.it4045npsproject.dto.Park;
+
+import java.util.List;
+
+/**
+ * Contains methods for business logic related to Parks
+ */
+public interface IParkService {
+    /**
+     * Save a park
+     * @param park - new park to be saved
+     * @return - newly saved park
+     */
+    Park save(Park park);
+
+    /**
+     * Delete a park
+     * @param park - park to be deleted
+     */
+    void delete(Park park);
+
+    /**
+     * Find a park with a given ID
+     * @param id - ID of a park
+     * @return - park
+     */
+    Park findParkById(int id);
+
+    /**
+     * Find a park in a specific location
+     * @param location - location
+     * @return - park
+     */
+    Park findParkByLocation(String location);
+
+    /**
+     * Add a favorite park
+     * @param park - park to be added as a favorite
+     */
+    void saveFavoritePark(Park park);
+
+    /**
+     * Return list of parks with the same activity
+     * @param activity - activity
+     * @return - list of parks
+     */
+    List<Park> findParksByActivity(String activity);
+}
```

Comment:
I also opted to create a service interface for Parks. Though the methods are identical to the DAO interface, once implemented, the service methods will call the DAO methods to talk with the database. It also allows you to add any business logic related methods in the future.

Source: https://www.capitalone.com/tech/software-engineering/10-microservices-best-practices/